File: Server/DTOs/Requests/RoomRequest.cs
namespace Server.DTOs
{
    public class RoomRequest
    {
        public required string Username { get; set; }
        public required string RoomName { get; set; }
        public required string Topic { get; set; }
        public string? Language { get; set; }
        public string? Number { get; set; }
        public string? Difficulty { get; set; }

    }
}

File: client/src/models/RoomRequest.ts
import type { FormData } from "../features/HomePage";

export interface RoomRequest {
    Username: string;
    RoomName: string;
    Topic: string;
    Language: string;
    Number: string;
    Difficulty: string;
}

export function ToRoomRequest(username: string, data: FormData): RoomRequest {
    return {
        Username: username,
        RoomName: data.roomName,
        Topic: data.topic,
        Language: data.language,
        Number: data.number,
        Difficulty: data.difficulty,
    }
}